db/data_migration/20150407095340_convert_policies_to_html_publications.rb

```diff
@@ -0,0 +1,143 @@
+module PolicyMigrationHelpers
+  def self.build_html_attachment_body(policy, url_maker, alternative_format_provider)
+    callout = %|
+$CTA
+This is a copy of a document that stated a policy of the 2010 to 2015 Conservative and Liberal Democrat coalition government.
+The previous URL of this page was [#{url_maker.public_document_url(policy)}](#{url_maker.public_document_path(policy)})
+Current policies can be found at the GOV.UK [policies list](/government/policies).
+$CTA
+
+|
+
+    combined_body = callout + policy.body
+
+    supporting_pages = policy.supporting_pages.published
+    supporting_pages.each_with_index do |supporting_page, index|
+      combined_body += extract_supporting_page(supporting_page, index, alternative_format_provider)
+    end
+
+    return combined_body
+  end
+
+  def self.copy_associations!(policy, policy_paper)
+    [
+      :lead_organisations,
+      :supporting_organisations,
+      :topics,
+      :topical_events,
+      :nation_inapplicabilities,
+      :role_appointments,
+      :fact_check_requests,
+      :world_locations,
+      :related_documents,
+      :specialist_sectors,
+    ].each do |association|
+      policy_paper.send("#{association}=".to_sym, policy.send(association))
+    end
+  end
+
+  def self.extract_supporting_page(supporting_page, index, alternative_format_provider)
+    appendix_title = "Appendix #{index+1}: #{supporting_page.title}"
+
+    puts "-- Adding supporting page ##{supporting_page.id}: as #{appendix_title}"
+
+    sp_body = %|
+
+###{appendix_title}
+
+$CTA
+This was a supporting detail page of the main policy document.
+$CTA
+
+#{supporting_page.body.gsub(/^(\s*[#]{2,})/, '\1#')}
+|
+
+    sp_body.gsub(/\[InlineAttachment:([0-9]+)\]/) do
+      if attachment = supporting_page.attachments[$1.to_i - 1]
+        "[#{attachment.title}](#{attachment.url})"
+      else
+        ""
+      end
+    end
+  end
+end
+
+gds_user = User.find_by!(name: "GDS Inside Government Team")
+url_maker = UrlMaker.new
+
+Policy.published.each do |policy|
+  puts %{Creating policy paper from policy ##{policy.id}}
+
+  title = "2010 to 2015 Conservative and Liberal Democrat coalition policy: #{policy.title}"
+  short_title = "2010 to 2015 coalition policy: #{policy.title}"
+
+  supporting_pages = policy.supporting_pages.published
+
+  alternative_format_provider = policy.lead_organisations.first
+
+  html_attachment = HtmlAttachment.new(
+    title: title,
+    slug: short_title.parameterize,
```

Comment:
It doesn't appear like this works - the slug is generated with the full title rather than the short one:

![screen shot 2015-04-14 at 12 40 01](https://cloud.githubusercontent.com/assets/3687/7135884/1412013a-e2a3-11e4-873a-10d1c4d7e71a.png)